File: SpringBootAsync/src/main/java/com/example/service/BasicService.java
package com.example.service;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Service;

@Service
public class BasicService {

	private static final Logger logger = LoggerFactory.getLogger(BasicService.class);

	@Async
	public void onAsync() throws InterruptedException {
		Thread.sleep(1000);
		logger.info("onAsync");
	}

	public void onSync() throws InterruptedException {
		Thread.sleep(1000);
		logger.info("onSync");
	}
}

File: SpringBootAsync/src/main/kotlin/com/example/kotlin/service/BasicService.kt
package com.example.kotlin.service

import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Service

@Service
class BasicService {

    companion object {
        private val logger = LoggerFactory.getLogger(BasicService::class.java)
    }

    @Async
    fun onAsync() {
        Thread.sleep(1000)
        logger.info("onAsync")
    }

    fun onSync() {
        Thread.sleep(1000)
        logger.info("onSync")
    }
}
